src/status_im/ui/screens/discover/all_dapps/views.cljs

```diff
@@ -0,0 +1,67 @@
+(ns status-im.ui.screens.discover.all-dapps.views
+  (:require-macros [status-im.utils.views :refer [defview letsubs]])
+  (:require [cljs.pprint :as pprint]
+            [re-frame.core :as re-frame]
+            [status-im.components.react :as react]
+            [status-im.components.chat-icon.screen :as chat-icon]
+            [status-im.components.carousel.carousel :as carousel]
+            [status-im.ui.screens.discover.components.views :as discover-components]
+            [status-im.ui.screens.discover.styles :as styles]
+            [status-im.utils.platform :as platform]
+            [status-im.i18n :as i18n]
+            [status-im.components.toolbar-new.view :as toolbar]))
+
+(defn dapp-item [dapp]
+  (let [{:keys [name photo-path dapp?]} dapp
+        ;; TODO(oskarth): Refactor this to be in discover styles, just uppercase
+        item-style (get-in platform/platform-specific [:component-styles :discover :item])]
+    [react/view
+     [react/view styles/dapps-list-item
+      [react/view styles/dapps-list-item-second-row
+      [react/view styles/dapps-list-item-name-container
+       [react/view (merge styles/dapps-list-item-avatar-container
+                          (:icon item-style))
+        ;; TODO(oskarth): Get rid of badge on icon
+        [react/view (chat-icon/contact-icon-contacts-tab dapp)]]
+       [react/text {:style           styles/dapps-list-item-name
+                    :font            :medium
+                    :number-of-lines 1}
+        name]]]]]))
+
+(defn preview [dapps]
+  [react/view styles/dapp-preview-container
+   ;; TODO(oskarth): Refactor to shorter form
+   [discover-components/title
+    :t/dapps
+    :t/all
+    #(re-frame/dispatch [:navigate-to :discover-all-dapps])]
+   (if (seq dapps)
+     ;; TODO(oskarth): Make this carousel show more dapps at a time
+     [carousel/carousel {:pageStyle styles/carousel-page-style
+                         :backgroundColor "white"
+                         :gap       4
+                         :sneak     4
+                         :scrollThreshold 20
+                         :count     (count dapps)}
+      (for [[_ dapp] dapps]
+        ^{:key (str (:name dapp))}
+        [react/view styles/dapp-preview-content
+         [dapp-item dapp]])]
+     [react/text (i18n/label :t/none)])])
+
+(defview main []
```

Comment:
Maybe a less generic name? I know there is no main method in cljs but that's the first thing I think of when I see this.

Reply:
@goranjovic what would you prefer? To me it is the main fn of all_dapps namespace, to avoid duplication like `all-dapps/all-dapps`. Happy to change it though.

Reply:
Yeah, I don't really like the duplication either. Keep it that way, we'll figure something out.